File: algorithms/java/binary/search/LC_0268_MissingNumber.java
// https://leetcode.com/problems/missing-number/description/

package binary.search;

import java.util.Arrays;

public class LC_0268_MissingNumber {
    public int missingNumber(int[] nums) {
        Arrays.sort(nums);
        int l = 0;
        int r = nums.length - 1;

        while (l <= r) {
            int m = l + (r - l) / 2;

            if (m == nums[m]) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }

        return l;
    }
}

File: algorithms/python3/binary.search/LC_0268_MissingNumber.py
# https://leetcode.com/problems/missing-number/description/

class Solution:
    def missingNumber(self, nums: list[int]) -> int:
        nums.sort()

        left, right = 0, len(nums)

        while left < right:
            mid = (left + right) // 2
            if nums[mid] == mid:
                left = mid + 1
            else:
                right = mid

        return left